signer/nostr-browser-signer-proxy/src/lib.rs

```diff
@@ -169,7 +170,19 @@ pub struct BrowserSignerProxy {
     shutdown: Arc<Notify>,
 }
 
-// TODO: use atomic-destructor to automatically shutdown this when all instances are dropped
+impl Drop for BrowserSignerProxy {
+    fn drop(&mut self) {
+        self.shutdown.notify_one();
+        self.shutdown.notify_waiters();
+    }
+}
+
+impl AtomicDestroyer for BrowserSignerProxy {
```

Comment:
The `AtomicDestroyer` trait isn't enough here, we have to use the `AtomicDestructor<T>` struct inside the `BrowserSignerProxy`. This requires to add a `BrowserSignerProxyInner` private struct and move some methods to it. It's important that when the stuff is cloned to be passed to the tasks, it's not cloned the `AtomicDestructor` otherwise it will not shutdown.